File: modules/gateway/src/main/java/com/gateway/gateway/healthcheck/HealthCheck.java
package com.gateway.gateway.healthcheck;

import com.common.result.type.CommonResult;
import com.common.web.response.NoDataView;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@Slf4j
@RestController
public class HealthCheck {

    @GetMapping("/health-check")
    public NoDataView healthCheck() {
        return NoDataView.builder()
                .result(CommonResult.REQUEST_SUCCESS)
                .build();
    }
}

File: modules/gateway/src/main/kotlin/com/media/bus/gateway/healthcheck/HealthCheck.kt
package com.media.bus.gateway.healthcheck

import com.media.bus.common.web.response.ApiResponse
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/gateway")
class HealthCheck {

    private val log = LoggerFactory.getLogger(HealthCheck::class.java)

    @GetMapping("/health-check")
    fun healthCheck(): ApiResponse<Unit?> {
        return ApiResponse.success()
    }
}
